### PR TITLE
posix: sysconf: add a "small" implementation option

### PR DESCRIPTION
**Pull request by cfriedt**
Previously, `sysconf()` was only available as a macro (i.e. the "extra-small" option).

That has the advantage of being compile-time constant, optimized for both space and speed. One disadvantage is that a querying an `_SC_` value that was invalid or unsupported would result in a compile error.

Provide a "small" implementation of `sysconf()` (via Kconfig choice) as a normal addressable function.

See also https://github.com/zephyrproject-rtos/zephyr/issues/56670

### CODE REVIEW

**Comment by cfriedt** [comment]
Was hoping to tag @awojasinski for review here as well.

File: include/zephyr/posix/unistd.h [hunk 1]
```diff
@@ -99,6 +99,7 @@ extern "C" {
 #define _POSIX2_C_BIND			  _POSIX_VERSION
 #define _POSIX2_C_DEV			  (-1L)
 #define _POSIX2_CHAR_TERM		  (-1L)
+#define _POSIX2_DELAYTIMER_MAX    (-1L)
```
Comment (ycsin) [inline include/zephyr/posix/unistd.h:102]:
```suggestion
#define _POSIX2_DELAYTIMER_MAX		  (-1L)
```

**Review by ycsin — DISMISSED** [review]
LGTM

File: include/zephyr/posix/unistd.h [hunk 2]
```diff
@@ -99,6 +99,7 @@ extern "C" {
 #define _POSIX2_C_BIND			  _POSIX_VERSION
 #define _POSIX2_C_DEV			  (-1L)
 #define _POSIX2_CHAR_TERM		  (-1L)
+#define _POSIX2_DELAYTIMER_MAX	  (-1L)
```
Comment (cfriedt) [inline include/zephyr/posix/unistd.h:102]: This is actually fixed but just renders oddly.

**Review by ycsin — APPROVED** [review]

**Review by awojasinski — APPROVED** [review]
LGTM

**Review by jukkar — APPROVED** [review]

**Pull request merged**